x-pack/plugins/security_solution/public/common/components/events_viewer/index.tsx

```diff
@@ -170,78 +215,402 @@ const StatefulEventsViewerComponent: React.FC<Props> = ({
+                              disabledCellActions={FIELDS_WITHOUT_CELL_ACTIONS}
+                              id={tableId}
+                              loadPage={loadPage}
+                              renderCellValue={renderCellValue}
+                              rowRenderers={rowRenderers}
+                              totalItems={totalCountMinusDeleted}
+                              bulkActions={bulkActions}
+                              fieldBrowserOptions={fieldBrowserOptions}
+                              defaultCellActions={defaultCellActions}
+                              hasCrudPermissions={hasCrudPermissions}
+                              filters={filters}
+                              leadingControlColumns={transformedLeadingControlColumns}
+                              pagination={{
+                                pageIndex: pageInfo.activePage,
+                                pageSize: itemsPerPage,
+                                pageSizeOptions: itemsPerPageOptions,
+                                onChangeItemsPerPage,
+                                onChangePage,
+                              }}
+                            />
+                          )}
+                          {tableView === 'eventRenderedView' && (
+                            <EventRenderedView
+                              events={nonDeletedEvents}
+                              leadingControlColumns={transformedLeadingControlColumns}
+                              pagination={{
+                                pageIndex: pageInfo.activePage,
+                                pageSize: itemsPerPage,
+                                totalItemCount: totalCountMinusDeleted,
+                                pageSizeOptions: itemsPerPageOptions,
+                                showPerPageOptions: true,
+                              }}
+                              rowRenderers={rowRenderers}
+                              scopeId={tableId}
+                              onChangePage={onChangePage}
+                              onChangeItemsPerPage={onChangeItemsPerPage}
+                              additionalControls={alertBulkActions}
+                              unitCountText={unitCountText}
+                            />
+                          )}
+                        </StatefulEventContext.Provider>
+                      </ScrollableFlexItem>
+                    </FullWidthFlexGroupTable>
+                  )}
+                </EventsContainerLoading>
+              </TableContext.Provider>
+            )}
+          </StyledEuiPanel>
         </InspectButtonContainer>
       </FullScreenContainer>
       {DetailsPanel}
     </>
   );
 };
 
-export const StatefulEventsViewer = React.memo(StatefulEventsViewerComponent);
+const mapDispatchToProps = {
+  clearSelected: dataTableActions.clearSelected,
+  setSelected: dataTableActions.setSelected,
+};
+
+const connector = connect(undefined, mapDispatchToProps);
```

Comment:
Is it possible to remove the `connected` and just use `useDispatch` for the `onRowSelected` and `onSelectPage`? I don't think we get any benefit from `connect` over hooks when were not actually pulling in any state

Reply:
We still need that `connect` for making checkboxes working. I didn't dig into details of its implementation, but we can revisit existing code in the follow up issue.